presto-docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,164 @@
+by response headers, and supplied in subsequent requests, just like browser cookies.
+
+====================================== =========================================================================================
+Request Header Name                    Description
+====================================== =========================================================================================
+``X-Presto-User``                      Specifies the session user; must be supplied with every
+                                       request to ``/v1/statement``.
+``X-Presto-Source``                    For reporting purposes, this supplies the name of the software that submitted the query.
+``X-Presto-Catalog``                   The catalog to be used when running the query.  Set by response header
+                                       ``X-Presto-Set-Catalog``.
+``X-Presto-Schema``                    The schema to be used when running the query.  Set by response header
+                                       ``X-Presto-Set-Schema``.
+``X-Presto-Time-Zone``                 The timezone to be used when running the query, which by default is the timezone of
+                                       the Presto engine.
+``X-Presto-Language``                  The language to be used when running the query and formatting results.  The language
+                                       of the session can be set on a per-query basis using the ``X-Presto-Language``
+                                       HTTP header, or via the ``PrestoConnection.setLocale(Locale)`` method in the
+                                       JDBC driver.
+``X-Presto-Trace-Token``               Supplies a trace token to the Presto engine to help identify log lines that originate
+                                       with this query request.
+``X-Presto-Session``                   Supplies a comma-separated list of name=value pairs as session properties.
+                                       When the Presto client run a ``SET SESSION name=value`` query, the name=value pair
+                                       is returned in the ``X-Set-Presto-Session`` response header, and added to the client's
+                                       list of session properties.
+                                       If the response header ``X-Presto-Clear-Session`` is returned, its value
+                                       is the name of a session property that will be removed from the client's accumulated
+                                       list.
+``X-Presto-Role``                      Sets the catalog role that will be used in this request.  Set by response header
+                                       ``X-Presto-Set-Role``.
+``X-Presto-Prepared-Statement``        A comma-separated list of the name=value pairs, where the names are names of
+                                       previously prepared SQL statements, and the values are ???.
```

Comment:
Is the `???` a placeholder?

Reply:
Oops.  ??? changed to "keys that identify the executable form of the named prepared statements."